dandi/download.py

```diff
@@ -1122,19 +1123,36 @@ def _check_if_more_attempts_allowed(
             return None
         elif retry_after := exc.response.headers.get("Retry-After"):
             # playing safe
-            if not str(retry_after).isdigit():
-                # our code is wrong, do not crash but issue warning so
-                # we might get report/fix it up
-                lgr.warning(
-                    "%s - download failed due to response %d with non-integer"
-                    " Retry-After=%r: %s",
-                    path,
-                    exc.response.status_code,
-                    retry_after,
-                    exc,
-                )
-                return None
-            sleep_amount = int(retry_after)
+            retry_after = str(retry_after)
+            if retry_after.isdigit():
+                sleep_amount = int(retry_after)
+            else:
+                # else if it is a datestamp like "Wed, 21 Oct 2015 07:28:00 GMT"
+                # we could parse it and calculate how long to sleep
+                try:
+                    retry_after_date = parsedate_to_datetime(retry_after)
+                    current_date = datetime.now(UTC)
+                    difference = retry_after_date - current_date
+                    sleep_amount = int(difference.total_seconds())
+                    if sleep_amount < 0:
+                        lgr.warning(
+                            "%s - date in Retry-After=%r is in the past (current is %r)",
+                            path,
+                            retry_after,
+                            current_date,
+                        )
+                except ValueError as exc_ve:
+                    # our code or response is wrong, do not crash but issue warning
+                    # and continue with "default" sleep logic
+                    lgr.warning(
+                        "%s - download failed due to response %d with non-integer or future date"
+                        " Retry-After=%r: %s with %s upon converting assuming it is a date",
+                        path,
+                        exc.response.status_code,
+                        retry_after,
+                        exc,
+                        exc_ve,
+                    )
```

Comment:
```suggestion
                try:
                    retry_after_date = parsedate_to_datetime(retry_after)
                except (ValueError, TypeError) as exc_ve:
                    # our code or response is wrong, do not crash but issue warning
                    # and continue with "default" sleep logic
                    lgr.warning(
                        "%s - download failed due to response %d with non-integer or future date"
                        " Retry-After=%r: %s with %s upon converting assuming it is a date",
                        path,
                        exc.response.status_code,
                        retry_after,
                        exc,
                        exc_ve,
                    )
                else:
                    current_date = datetime.now(timezone.utc)
                    difference = retry_after_date - current_date
                    sleep_amount = int(difference.total_seconds())
                    if sleep_amount < 0:  # todo: `sleep_amount < 0` may not be an error. It the server set a `Retry-After` datetime extremely close and the network is slow, the client may receive the response after the `Retry-After` datetime.
                        lgr.warning(
                            "%s - date in Retry-After=%r is in the past (current is %r)",
                            path,
                            retry_after,
                            current_date,
                        )
```

This suggestion make the try clause 
1. more targeted
2. catches `TypeError`s from `parsedate_to_datetime`. (Try `parsedate_to_datetime('Wed, 21 O3t 2015 07:280 GMT')`)
3. adds a note with a todo comment regarding `sleep_amount < 0`